go/engine/kex2_provisionee.go

```diff
@@ -288,6 +290,21 @@ func (e *Kex2Provisionee) handleDidCounterSign(sig []byte, sdhBoxes []keybase1.S
 		return err
 	}
 
+	// remember how to update a user with the new links
+	e.localDelegateFn = func(u *libkb.User) (err error) {
```

Comment:
Yes, this is the main fix. The paper key generation bug is that the user object was stale so the paperkey signed the wrong sigchain end.